src/explosion.cpp

```diff
@@ -218,21 +219,24 @@ static void do_blast( const tripoint &p, const float power,
                 continue;
             }
 
-            // Up to 200% bonus for shaped charge
-            // But not if the explosion is fiery, then only half the force and no bonus
-            const float bash_force = !fire ?
-                                     force + ( 2 * force / empty_neighbors ) :
-                                     force / 2;
-            if( z_offset[i] == 0 ) {
-                // Horizontal - no floor bashing
-                g->m.bash( dest, bash_force, true, false, false );
-            } else if( z_offset[i] > 0 ) {
-                // Should actually bash through the floor first, but that's not really possible yet
-                g->m.bash( dest, bash_force, true, false, true );
-            } else if( !g->m.valid_move( pt, dest, false, true ) ) {
-                // Only bash through floor if it doesn't exist
-                // Bash the current tile's floor, not the one's below
-                g->m.bash( pt, bash_force, true, false, true );
+            if( bashed.count( dest ) != 0 ) {
```

Comment:
It looks like it should've been:

```suggestion
            if( bashed.count( dest ) == 0 ) {
```

I.e. if tiles was not bashed yet - bash it. Otherwise, no tiles except center point would ever be bashed as `bashed` collection is initialized with centeral point and only populated from inside of the this if statement.